### ISSUE TITLE
devicefarmUpload fail with NullPointerException on android library (aar) projects

### ISSUE THREAD

**Issue by danielesegato** [awslabs/aws-device-farm-gradle-plugin#14]
I've an android library project and am trying to test it using AWS Device Farm, when i execute it I get this error:

```
:my-library:devicefarmUpload FAILED
:my-library:devicefarmUpload (Thread[Task worker Thread 7,5,main]) completed. Took 0.889 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':my-library:devicefarmUpload'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':currency-converter-datalayer:devicefarmUpload'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:84)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:236)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:228)
        at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:61)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:228)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:215)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:77)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:58)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
Caused by: java.lang.NullPointerException
        at com.amazonaws.devicefarm.DeviceFarmUploader.upload(DeviceFarmUploader.java:64)
        at com.amazonaws.devicefarm.DeviceFarmServer.uploadApks(DeviceFarmServer.java:87)
        at com.android.build.gradle.internal.tasks.TestServerTask.sendToServer(TestServerTask.java:45)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:123)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:632)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:615)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:76)
        ... 20 more


BUILD FAILED
```

The configuration is really basic:

```gradle
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.3.0'
        classpath 'com.amazonaws:aws-devicefarm-gradle-plugin:1.2'
    }
}
```

and in the library project below the android library setup:

```gradle
apply plugin: 'devicefarm'

devicefarm {
    projectName "Bootstrap Android"
    devicePool "Last Nexus/Pixel"
    authentication {
        accessKey "[redacted]"
        secretKey "[redacted]"
    }

    // optional block, radios default to 'on' state, all parameters optional
    devicestate {
    }

    instrumentation {
        filter "my.package"
    }
}
```

The exact same configuration succeed on an application (apk).

`connectedAndroidTest`s are executed just fine on the library when i use my own device or an emulator.

**Comment by andreas-marschke** [comment]
This is due to how Android Library Tests are set up. I came across the same issue as well and fixed it by patching the gradle plugin. I'll upload a branch and reference it here.

**Comment by andreas-marschke** [comment]
@danielesegato #20 has been merged. I hope we'll get a new release soon. In the meantime can you validate against your usecase that the plugin works when you build it from source? 

Thanks!

**Comment by bradley-curran** [comment]
This fix is in version 1.3. Closing.
